### ISSUE TITLE
Can't add UWP apps to groups, get "Catastrophic failure" errors

### ISSUE THREAD

**Issue by Neamow** [tjackenpacken/taskbar-groups#69]
Hello,

when I try to add a UWP app (e.g. Skype, Groove, etc...) to a group, I just get a popup saying "Catastrophic failure", and nothing happens.

Any way to fix this? Thanks!

**Comment by PikeNote** [comment]
Any other information when it says "Catastrophic Failure"?

I had tested it locally and it seems to work.
How are you adding the UWP app? Through dragging it from the start menu directly or creating a shortcut?
The only bug I have noticed currently is Groove Music having a weird name when being imported.

**Comment by Neamow** [comment]
I have no idea what you mean by dragging it from the start menu, the groups settings window doesn't allow drag and drop of icons for me.

Additionally since it's impossible to navigate to the install folder for UWP apps (C:/Program Files/WindowsApps, it gives you a permission error even if you're the administrator), I have to create a shortcut on the desktop.

That's what I tried using when adding programs to the groups, but I just get the "Catastrophic failure" error.

You say you tested it and it works for you - how are you doing it?

**Comment by PikeNote** [comment]
You can also pin the Windows UWP app to the start menu and drag and drop it into the application from there.

The shortcut method seems to work aswell. Is there any other information to the Catastrophic Failure or is it a message box saying that?

**Comment by Neamow** [comment]
Sorry I tried dragging but it simply doesn't work. There's a red icon and it doesn't allow me to drag and drop.

Unfortunately no, it doesn't say anything else.

I added a short screen recording, I think that'll explain it all better than many words: https://www.youtube.com/watch?v=Z-g__noJhs8.

**Comment by PikeNote** [comment]
Hmm, seems like you are using an older version of Taskbar groups. 
Try the latest release from here. (https://github.com/tjackenpacken/taskbar-groups/releases)

You need to drag the UWP from start to the + icon area where you click to add your shortcuts normally.

**Comment by Neamow** [comment]
Ah ok that may be it. I found this on [reddit ](https://www.reddit.com/r/Windows10/comments/kew389/i_made_a_free_application_that_lets_you_group/)and the download link in the comment there only went to the 0.1.4 version. Didn't notice there's a newer one.

However UWP apps still don't work for me in 0.2.0 either. It allows me to add them, yes, even drag-and-drop, but I get a popup with a weird path to the app. If I close it, it allows me to save the group, but shows the popup again. I close it again, then add the group to the taskbar, but when I click on it I get a .NET framework error.

![Clipboard01](https://user-images.githubusercontent.com/8394186/103812320-a75d7580-505e-11eb-9659-9f701983176e.png)

![Clipboard02](https://user-images.githubusercontent.com/8394186/103812329-a9bfcf80-505e-11eb-94d6-b49ecc8c40b5.png)

![Clipboard03](https://user-images.githubusercontent.com/8394186/103812337-acbac000-505e-11eb-914a-da4f6f5e408e.png)

**Comment by PikeNote** [comment]
Yeah I see the issue now. Seems like the Release version had some odd bugs that aren't reproduced on the Debug version. I will look into it.

**Comment by PikeNote** [comment]
I think I found the root issue?
[Release.zip](https://github.com/tjackenpacken/taskbar-groups/files/5778324/Release.zip)
Try this experimental release.

I am so dead inside after experimenting and then it behaving just a bit differently each time for no reason at all. 😭
That and somehow a for loop was causing an issue that was added in last second as a fix.

**Comment by Neamow** [comment]
Hey man, don't sweat it. That's software development for you ;) patch one bug, another two will pop up.

I've tried the release and I'm glad to say I've got no errors and no issues adding UWP or any other apps. The only thing wrong is that the order of icons in the group once on the taskbar - it's launching the apps in the same order as set in the config window, they just have wrong icons.

Thanks!

**Comment by PikeNote** [comment]
Will be fixed in the next release which may be hopefully soon.

**Comment by JohnLGalt** [comment]
I've not been able to get any sort of UWP (Like Windows Mail, for example) to add to TB Group.  I'm using the test for the Google Chrome Fix, if that makes a difference.

**Comment by PikeNote** [comment]
It has been fixed in the #71 pull, but it hasn't been merged yet. I can post a preview build of the latest from my repo with some newer changes in a bit.

**Comment by JohnLGalt** [comment]
Nah, I can wait.  It's one single UWP app, the Mail app, and with everything else working fine, I'm in no rush.  I just uninstalled BINS again because I now remember why it was not conducive to using - after 1803 came out it had a bad habit of auto hiding itself when you're trying to start an app in it,m making it frustrating beyond belief.  So after a week I uninstalled it and then saw the fix for Chrome so gave TbG another shot - and presto, I'm good again.
